outbound/mx_lookup.js

```diff
@@ -33,8 +35,10 @@ exports.lookup_mx = function lookup_mx (domain, cb) {
         }
         else if (addresses && addresses.length) {
             for (let i=0,l=addresses.length; i < l; i++) {
-                const mx = wrap_mx(addresses[i]);
-                mxs.push(mx);
+                if (obc.cfg.local_mx_ok || !net_utils.is_local_ip(addresses[i].exchange)) {
```

Comment:
Sorry for being late to the party but addresses[i].exchange gives me hostname not IP

Reply:
Thanks for the feedback @analogic. 

I just tested it and you're right, I get the hostname too, whereas `net_utils.is_local_ip` seems to expect an ip. 

The change solves #2809 anyway because (from what I understand) the issue was addressing cases where the MX record was returning a local IP (not a hostname), such as with domain yeah.com:

```
$ dig MX yeah.com
...
yeah.com.		534	IN	MX	1000 0.0.0.0.
...
```

In that case `addresses = [ { exchange: '0.0.0.0', priority: 1000 } ]` and the fix does work.

The fix would fail for an MX record resolving to a local ip. Is this a legitimate use case? If yes, perhaps we could use `net_utils.get_ips_by_host` to resolve `addresses[i].exchange` to an ip? I'm wondering if this can badly impact performance though. I'm not knowledgeable enough on Haraka to evaluate this.

Reply:
use `net.isIP` to define if .exchange is an IP or not. When its not an IP, using `net_utils.get_ips_by_host` to resolve should be fine.

Reply:
I've made two PRs as a fix attempts:
- https://github.com/haraka/Haraka/pull/3002
- https://github.com/haraka/haraka-net-utils/pull/63